android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt

```diff
@@ -80,7 +80,14 @@ fun List<ActionConfig>.handleClickEvent(
               NavigationArg.RESOURCE_CONFIG to actionConfig.resourceConfig,
               NavigationArg.PARAMS to interpolatedParams.toTypedArray(),
             )
-          navController.navigate(MainNavigationScreen.Profile.route, args)
+          navController.navigate(
+            resId = MainNavigationScreen.Profile.route,
+            args = args,
+            navOptions =
+              navController.currentDestination?.id?.let {
+                navOptions(resId = it, inclusive = actionConfig.popNavigationBackStack == true)
+              },
+          )
```

Comment:
This affects back navigation from Profile to Register. E.g. Move from Register -> Profile(2) -> Profile (2). Backpressing from Profile(2) moves to Register instead of Profile(1).

Reply:
oh, you're right

Reply:
fixed w my new commit